### PR TITLE
Add pagination modules

### PR DESCRIPTION
**Pull request by saivickna**
Why?

Pagination logic can be shared amount many nodule-graphql applications

### CODE REVIEW

File: src/modules/paging/concurrency.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import throat from 'throat';
+
+export default function concurrentPaginate(promises, concurrencyLimit = 1) {
+    const throatWithPromise = throat(Promise);
+    const funneledThroat = throatWithPromise(concurrencyLimit);
+    return Promise.all(
+        promises.map(promise => funneledThroat(() => promise)),
+    );
+}
```
Comment (saivickna) [inline src/modules/paging/concurrency.js]: Might move concurrency out of paging folder

File: src/index.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ export {
     UnprocessableEntity,
 } from './errors';
 export { createResolver, getResolver } from './resolver';
+export * from './modules';
```
Comment (jessemyers) [inline src/index.js]: Can we avoid `*`?

File: src/modules/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from './paging';
```
Comment (jessemyers) [inline src/modules/index.js]: Same, would rather have explicit imports.

File: src/modules/paging/allItems.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { flatten, range } from 'lodash';
+import { MaxLimitReached } from '../../errors';
+import concurrentPaginate from './concurrency';
+
+export default async function allItems(
```
Comment (jessemyers) [inline src/modules/paging/allItems.js]: Naming question: should `allItems` be `all`?

Our other operations are already in terms of items but aren't named as such.

**Pull request merged**